docs/v0.6/en/user-guide/client-libraries/go.md

```diff
@@ -36,129 +37,232 @@ import (
+
+// Insert example data
+// NOTE: The arguments must be in the same order as the columns in the defined schema: host, ts, cpu_user, cpu_sys
+err = cpuMetric.AddRow("127.0.0.1", time.Now(), 0.1, 0.12)
+if err != nil {
+    // Handle error appropriately
+}
 
-client, _ := greptime.NewClient(cfg)
 ```
 
 %}
 
+{template create-rows%
 
+```go
+cpuMetric, err := table.New("cpu_metric")
+if err != nil {
+    // Handle error appropriately
+}
+cpuMetric.AddTagColumn("host", types.STRING)
+cpuMetric.AddTimestampColumn("ts", types.TIMESTAMP_MILLISECOND)
+cpuMetric.AddFieldColumn("cpu_user", types.FLOAT)
+cpuMetric.AddFieldColumn("cpu_sys", types.FLOAT)
+err = cpuMetric.AddRow("127.0.0.1", time.Now(), 0.1, 0.12)
+if err != nil {
+    // Handle error appropriately
+}
 
-{template row-object%
+memMetric, err := table.New("mem_metric")
+if err != nil {
+    // Handle error appropriately
+}
+memMetric.AddTagColumn("host", types.STRING)
+memMetric.AddTimestampColumn("ts", types.TIMESTAMP_MILLISECOND)
+memMetric.AddFieldColumn("mem_usage", types.FLOAT)
+err = memMetric.AddRow("127.0.0.1", time.Now(), 112)
+if err != nil {
+    // Handle error appropriately
+}
+```
+
+%}
 
-The Go ingester SDK uses `Series` to represent a row data item. Multiple `Series` can be added to a `Metric` object and then written to GreptimeDB.
+{template insert-rows%
+
+```go
+resp, err := cli.Write(context.Background(), cpuMetric, memMetric)
+if err != nil {
+    // Handle error appropriately
+}
+log.Printf("affected rows: %d\n", resp.GetAffectedRows().GetValue())
+```
 
 %}
 
+{template streaming-insert%
+
+To insert data using the streaming API, you must first create a stream client.
+
+```go
+cfg := config.New("<host>").
+    WithAuth("<username>", "<password>").
+    WithDatabase(database)
+streamClient, err := client.NewStreamClient(cfg)
+```
 
-{template create-a-rows%
+Then send the data to the server.
 
 ```go
-seriesHost1 := greptime.Series{}
-seriesHost1.AddStringTag("host", "host1")
-seriesHost1.AddFloatField("cpu", 0.90)
-seriesHost1.SetTimestamp(time.Now())
+err := streamClient.Send(context.Background(), cpuMetric, memMetric)
+if err != nil {
+    // Handle error appropriately
+}
 ```
 
 %}
 
-{template create-rows%
+{template update-rows%
 
 ```go
-seriesHost1 := greptime.Series{}
-seriesHost1.AddStringTag("host", "host1")
-seriesHost1.AddFloatField("cpu", 0.90)
-seriesHost1.SetTimestamp(time.Now())
-
-seriesHost2 := greptime.Series{}
-seriesHost2.AddStringTag("host", "host2")
-seriesHost2.AddFloatField("cpu", 0.70)
-seriesHost2.SetTimestamp(time.Now())
-
-metric := greptime.Metric{}
-metric.AddSeries(seriesHost1, seriesHost2)
+ts := time.Now()
+_ = cpuMetric.AddRow("127.0.0.1", ts, 0.1, 0.12)
+// insert a row data
+resp, _ := cli.Write(context.Background(), cpuMetric)
+
+// update the row data
+// The same tag `127.0.0.1`
+// The same time index `ts`
+// The new value: cpu_user = `0.80`, cpu_sys = `0.11`
+_ = cpuMetric.AddRow("127.0.0.1", ts, 0.80, 0.11)
```

Comment:
@yuanbohan I'm not sure this is the best practice.

Reply:
1. update operation is not recommended in TSDB, thus update-rows is not that important, which may be omitted here @fengjiachun 
2. ingester-go does not implement or test update or delete operation so far

Reply:
We consider `update data` as a feature that [users concern](https://docs.greptime.com/user-guide/concepts/features-that-you-concern#does-greptimedb-support-updates), so the code documentation should explain how to update data.

Reply:
I have not test the update logic. Is it ok to leave a `Not Implemented` here?